### PR TITLE
Add CLI implementation, and clean up services.

### PR DESCRIPTION
**Pull request by AliUnipal**
- Cleanup the context (To t.Context) in the unit tests.
- Add new user unit test for empty image url.

### CODE REVIEW

File: cmd/cli/main.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+	chatSvc    chatService
+	userSvc    userService
+	messageSvc messageService
+}
+
+func getCreateUserInput(args []string) (usersvc.CreateUserInput, error) {
+	var userInput usersvc.CreateUserInput
+
+	userFlags := flag.NewFlagSet("createUser", flag.ExitOnError)
+	firstName := userFlags.String("firstName", "", "First name (Required)")
```
[the remark this answers is not in the document]
Comment (AliUnipal) [reply cmd/cli/main.go:48]: Much cleaner.

File: internal/service/chatsvc/repo/inmemchatrepo/repo.go [hunk 2]
```diff
@@ -5,36 +5,61 @@ import (
...
 func (r *repository) CreateChat(ctx context.Context, in repo.CreateChatInput) error {
 	ids := []string{in.CurrentUserID.String(), in.OtherUserID.String()}
 	slices.Sort(ids)
 	id := strings.Join(ids, "|")
-	if _, ok := r.chats[id]; ok {
+	if _, ok := r.Chats[id]; ok {
 		return errors.New("chat already exists")
 	}
+	if in.CurrentUserID == in.OtherUserID {
+		return errors.New("user ids are matching, you cannot create a chat with yourself!")
```
[the remark this answers is not in the document]
Comment (AliUnipal) [reply internal/service/chatsvc/repo/inmemchatrepo/repo.go:61]: We do self-chat like WhatsApp y3ni? 😂

File: internal/service/chatsvc/repo/model.go [hunk 3]
```diff
@@ -2,29 +2,26 @@ package repo
...
+	CurrentUser User      `json:"current_user"`
+	OtherUser   User      `json:"other_user"`
 }
 
-type Message struct {
-	ID        uuid.UUID
-	SenderID  uuid.UUID
-	Content   []byte
-	Timestamp time.Time
+// TODO: Ask if this is wrong todo
```
[the remark this answers is not in the document]
Comment (AliUnipal) [reply internal/service/chatsvc/repo/model.go:21]: The question is if it's wrong to put the struct as export, because kind of this is specifically made for the inmem repo. However, I put the struct in the inmemchat repo file as not exported, and I forgot to cleanup this 😅

File: internal/service/msgsvc/repo/inmemmessagerepo/repo.go [hunk 4]
```diff
@@ -5,31 +5,47 @@ import (
...
+	snapper  *snapper.FileSnapper[messages]
 }
 
 func (r *repository) CreateMessage(ctx context.Context, in repo.CreateMessageInput) error {
 	c, err := r.chatRepo.GetChat(ctx, in.ChatID)
 	if err != nil {
 		return err
 	}
-	if c.CurrentUser.ID != in.SenderID || c.OtherUser.ID != in.SenderID {
+	if c.CurrentUser.ID != in.SenderID && c.OtherUser.ID != in.SenderID {
```
[the remark this answers is not in the document]
Comment (AliUnipal) [reply internal/service/msgsvc/repo/inmemmessagerepo/repo.go:48]: 3shank I`ll do it

File: user_data.json [hunk 5]
```diff
@@ -0,0 +1 @@
+{"1a72c0b2-4db0-47e8-bf84-18adbd4ce2fc":{"id":"1a72c0b2-4db0-47e8-bf84-18adbd4ce2fc","image_url":"http://test.com","first_name":"potato","last_name":"meow","username":"+97312345678"},"7496cf40-f375-4c41-a28a-00ea092ebdf1":{"id":"7496cf40-f375-4c41-a28a-00ea092ebdf1","image_url":"http://test.com","first_name":"potato","last_name":"meow","username":"+97312345678"},"9caa016c-55cd-4960-a4ba-100c8e109285":{"id":"9caa016c-55cd-4960-a4ba-100c8e109285","image_url":"http://test.com","first_name":"potato","last_name":"meow","username":"+97312345678"},"cc774f97-2b5d-4652-9b33-b5a2f97b5646":{"id":"cc774f97-2b5d-4652-9b33-b5a2f97b5646","image_url":"http://test.com","first_name":"potato","last_name":"meow","username":"+97312345678"}}
```
Comment (almahoozi) [inline user_data.json:1]: You probably shouldn't commit this, but sure why not 😊 good for a starting point when testing.
